src/test/FuzzerImpl.cpp

```diff
@@ -69,17 +71,137 @@ getShortKey(PublicKey const& pk)
     return getShortKey(pk.ed25519());
 }
 
+uint8_t
+getShortKey(ClaimableBalanceID const& balanceID)
+{
+    return balanceID.v0()[0];
+}
+
+uint8_t
+getShortKey(LedgerKey const& key)
+{
+    switch (key.type())
+    {
+    case ACCOUNT:
+        return getShortKey(key.account().accountID);
+    case OFFER:
+        return getShortKey(key.offer().sellerID);
+    case TRUSTLINE:
+        return getShortKey(key.trustLine().accountID);
+    case DATA:
+        return getShortKey(key.data().accountID);
+    case CLAIMABLE_BALANCE:
+        return getShortKey(key.claimableBalance().balanceID);
+    }
+    throw std::runtime_error("Unknown key type");
+}
+
+void
+setAssetCode4(AssetCode4& code, int digit)
+{
+    strToAssetCode(code, "Ast" + std::to_string(digit));
+}
+
 // constructs an Asset structure for an asset issued by an account id comprised
-// of bytes reading [0,0,...,i] and an alphanum4 asset code of "Ast + i"
+// of bytes reading [0,0,...,digit] and an alphanum4 asset code of "Ast + digit"
 Asset
-makeAsset(int i)
+makeAsset(int digit)
 {
     Asset asset;
     asset.type(ASSET_TYPE_CREDIT_ALPHANUM4);
-    strToAssetCode(asset.alphaNum4().assetCode, "Ast" + std::to_string(i));
-    setShortKey(asset.alphaNum4().issuer, i);
+    setAssetCode4(asset.alphaNum4().assetCode, digit);
+    setShortKey(asset.alphaNum4().issuer, digit);
     return asset;
 }
+
+AssetCode
+makeAssetCode(int digit)
+{
+    AssetCode code;
+    code.type(ASSET_TYPE_CREDIT_ALPHANUM4);
+    setAssetCode4(code.assetCode4(), digit);
+    return code;
+}
+
+void
+generateStoredLedgerKeys(StoredLedgerKeys& keys)
+{
+    size_t constexpr NUM_UNVALIDATED_LEDGER_KEYS = 0x08U;
+    static_assert(NUM_UNVALIDATED_LEDGER_KEYS < NUM_STORED_LEDGER_KEYS,
+                  "No room for valid ledger keys");
+
+    auto const firstUnvalidatedLedgerKey =
+        NUM_STORED_LEDGER_KEYS - NUM_UNVALIDATED_LEDGER_KEYS;
+
+    for (auto index = 0; index < firstUnvalidatedLedgerKey; ++index)
+    {
+        keys[index] =
+            LedgerEntryKey(LedgerTestUtils::generateValidLedgerEntry());
+    }
+
+    for (auto index = firstUnvalidatedLedgerKey; index < NUM_STORED_LEDGER_KEYS;
+         ++index)
+    {
+        size_t const entrySize = 3;
+        keys[index] = autocheck::generator<LedgerKey>()(entrySize);
+    }
+}
+
+void
+storeLedgerKey(StoredLedgerKeys& keys, LedgerKey const& key)
+{
+    auto byte = getShortKey(key);
+    keys[byte % NUM_STORED_LEDGER_KEYS] = key;
+}
+
+void
+storeLedgerKeyIDsForOp(StoredLedgerKeys& keys,
```

Comment:
I don't see `storeLedgerKeyIDsForOp` used anywhere.

Reply:
Yikes!  That's kind of critical!  It perished in a rebase.  It's meant to be called from `applySetupOperations()` (where the `keys` parameter comes from `mStoredLedgerKeys`).  Thanks!

Reply:
It's restored now.